llvm/test/SYCLLowerIR/printf_addrspace/diagnose_compile_time_unknown.ll

```diff
@@ -0,0 +1,78 @@
+;; This tests diagnostic emission upon compile-time-unknown format string.
+;; In this instance, a select instruction is given as the first argument
+;; to __spirv_ocl_printf - as a result, moving the function into the constant AS
+;; becomes impossible.
+
+;; The IR is based on the following SYCL code:
+;;
+;; #include <CL/sycl.hpp>
+;; using namespace sycl;
+;; int foo(int k) {
+;;   queue q;
+;;   buffer<int, 1> buf(k);
+;;   q.submit([&](handler &cgh) {
+;;     auto acc = buf.get_access(cgh);
+;;     cgh.single_task([=]() {
+;;       if (acc[0] == 0)
+;;         ext::oneapi::experimental::printf("String 0\n");
+;;       else
+;;         ext::oneapi::experimental::printf("String 1\n");
+;;     });
+;;   });
+;;   return 0;
+;; }
```

Comment:
Let's put this into a separate `Inputs/name.cpp` file

Reply:
Done in https://github.com/intel/llvm/pull/5306/commits/fe83fa8df95755380aafe6d5abab27eb5d4614e1.